org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java

```diff
@@ -140,31 +141,35 @@ private void doCutCopyWithImportsOperation() {
 
 					@Override
 					public Boolean apply(Clipboard clipboard) {
-						Map<Object,Transfer> payload = newLinkedHashMap();
-						payload.put(cbData, TRANSFER_INSTANCE);
-						
-						TextTransfer textTransfer = TextTransfer.getInstance();
-						String textData = (String) clipboard.getContents(textTransfer);
-						if (textData == null || textData.isEmpty()) {
-							// StyledText copied any data to ClipBoard
+						try {
+							Map<Object,Transfer> payload = newLinkedHashMap();
+							payload.put(cbData, TRANSFER_INSTANCE);
+							
+							TextTransfer textTransfer = TextTransfer.getInstance();
+							String textData = (String) clipboard.getContents(textTransfer);
+							if (textData == null || textData.isEmpty()) {
+								// StyledText copied any data to ClipBoard
+								return Boolean.FALSE;
+							}
+							payload.put(textData, textTransfer);
+							
+							RTFTransfer rtfTransfer = RTFTransfer.getInstance();
+							String rtfData = (String) clipboard.getContents(rtfTransfer);
+							if (rtfData != null && !rtfData.isEmpty()) {
+								payload.put(rtfData, rtfTransfer);
+							}
+							
+							List<Object> datas = newArrayList();
+							List<Transfer> dataTypes = newArrayList();
+							for (Entry<Object, Transfer> entry : payload.entrySet()) {
+								datas.add(entry.getKey());
+								dataTypes.add(entry.getValue());
+							}
+							clipboard.setContents(datas.toArray(), dataTypes.toArray(new Transfer[] {}));
```

Comment:
should we have a look at the error and if the e.code == DND.ERROR_CANNOT_SET_CLIPBOARD show a dialog?

Reply:
And what to present? Further Clipboard is platform specific.
I don't think that this is worth handling in more detail.

Reply:
have a look at org.eclipse.debug.internal.ui.viewers.model.VirtualCopyToClipboardActionDelegate.writeBufferToClipboard(StringBuffer) or org.eclipse.jdt.internal.ui.actions.CopyQualifiedNameAction.run()

Reply:
Good point. I was not ware that other actions do this also. I will adjust this accordingly.

Reply:
Updated.